dns/dns.go

```diff
@@ -0,0 +1,116 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/iam"
+
+	dns "github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2017-09-01/dns"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// A dns.ZonesClient manages DNS zones (as opposed to records within zones).
+func getZonesClient() dns.ZonesClient {
+	token, _ := iam.GetResourceManagementToken(iam.AuthGrantType())
+	zonesClient := dns.NewZonesClient(helpers.SubscriptionID())
+	zonesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	zonesClient.AddToUserAgent(helpers.UserAgent())
+	return zonesClient
+}
+
+func getRecordSetsClient() dns.RecordSetsClient {
+	token, _ := iam.GetResourceManagementToken(iam.AuthGrantType())
+	recordsClient := dns.NewRecordSetsClient(helpers.SubscriptionID())
+	recordsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	recordsClient.AddToUserAgent(helpers.UserAgent())
+	return recordsClient
+}
+
+// CreateZone creates a new DNS zone (the top-level mgmt object for DNS in Azure).
+func CreateZone(ctx context.Context, zoneName string) (dns.Zone, error) {
+	zonesClient := getZonesClient()
+	return zonesClient.CreateOrUpdate(
+		ctx,
+		helpers.ResourceGroupName(),
+		zoneName,
+		dns.Zone{
+			Location: to.StringPtr("global"),
```

Comment:
>global [](start = 27, length = 6)

I dont know enough about the DNS service, exactly what dooes `global` mean in this context? Can't zones be created on westus?

Reply:
Dns is a global service so it doesn't really make sense to have a zone in a region. Is the parameter required ? it might be needed for private zones as those will be bound to a region.